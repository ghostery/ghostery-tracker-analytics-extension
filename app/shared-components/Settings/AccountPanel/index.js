import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import AccountPanel from './AccountPanel.jsx';
import { login, register, resetPassword, sendMetrics } from '../../../store/messages/messageCreators';
import {
  openToast,
  closeToast,
  updateLoginStatus,
  updateRegisterStatus,
  updateResetPasswordStatus,
} from '../../../store/actions/UserInterfaceActions';

const mapStateToProps = state => ({
  messageCreators: { login, register, resetPassword, sendMetrics },
  loginFailed: state.UserInterfaceReducer.loginFailed,
  registerFailed: state.UserInterfaceReducer.registerFailed,
  resetPasswordFailed: state.UserInterfaceReducer.resetPasswordFailed,
  resetPasswordSent: state.UserInterfaceReducer.resetPasswordSent,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    openToast,
    closeToast,
    updateLoginStatus,
    updateRegisterStatus,
    updateResetPasswordStatus,
  }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(AccountPanel);
