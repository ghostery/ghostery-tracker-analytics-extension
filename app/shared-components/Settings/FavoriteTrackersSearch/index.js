import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import FavoriteTrackersSearch from './FavoriteTrackersSearch.jsx';
import { openToast } from '../../../store/actions/UserInterfaceActions';
import {
  searchBugDb, addStarredTracker, removeStarredTracker,
} from '../../../store/messages/messageCreators';

const mapStateToProps = state => ({
  bugDbSearchResults: state.UserInterfaceReducer.bugDbSearchResults,
  starredTrackerIds: state.SettingsReducer.localSettings.starredTrackerIds,
  showToasts: state.SettingsReducer.localSettings.showToasts,
  messageCreators: { searchBugDb, addStarredTracker, removeStarredTracker },
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ openToast }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(FavoriteTrackersSearch);
