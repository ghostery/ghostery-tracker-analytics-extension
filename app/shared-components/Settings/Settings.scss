@import '../../utils/scss/color-variables.scss';

$footer-height: 271px;

.Settings {
  width: 100%;
  height: inherit;
  font-weight: 600;
}

.Settings__main {
  overflow-y: auto;
  height: inherit;
}

.Settings__closeButton {
  position: relative;
  top: 10px;
  left: calc(100% - 20px);
  width: 10px;
  min-height: 11px;
  cursor: pointer;
}

.Settings__settingsTabsContainer {
  flex-grow: 1;
}

.Settings__settingsTabs {
  margin-top: 9px;
}

.Settings__blackLine {
  position: relative;
  top: 34px;
  background-color: $black;
  width: 30px;
  height: 1px;
}

.nav-item > .Settings__settingsTab.nav-link {
  margin-top: 8px;
  background-color: $medium-grey;
  box-sizing: border-box;
  outline: 0;
  border-bottom: 1px solid $black;
  padding: 6px 0 0;
  min-width: 120px;
  height: 27px;
  font-size: 10px;
  text-align: center;
  text-decoration: none;
  color: $white;
  cursor: pointer;

  &:hover {
    color: $white;
  }

  &:focus {
    color: $white;
  }

  &.active {
    margin-top: 7px;
    background-color: transparent;
    border: 1px solid $black;
    border-bottom: 0;
    height: 28px;
    color: $dark-grey;
  }
}

.Settings__tabPane {
  height: 100%;
  min-height: 373px;
}

.Settings__displaySettingsTabContent {
  margin: 17px 13px;
  width: 92%;
}

.Settings__headerText {
  margin: 0;
  font-weight: 700;
  font-size: 13px;
  color: $dark-grey;

  &.panelLayout {
    margin: 26px 0;
  }

  &.panelPalettes {
    margin: 29px 0 26px;
  }
}

.Settings__textToggleContainer {
  width: 100%;
}

.Settings__toggle {
  position: relative;
  cursor: pointer;

  .Settings__toggleSlider {
    position: relative;
    background-color: $opaque-background-light;
    border-radius: 5px;
    width: 26px;
    height: 10px;
    transition: background-color .2s;
  }

  .Settings__toggleButton {
    position: absolute;
    left: -1px;
    box-shadow: 0 .5px 2px $dark-grey;
    background-color: $off-white;
    border-radius: 8px;
    width: 16px;
    height: 16px;
    transition: background-color .2s, left .2s;
  }

  > p {
    margin-left: 8px;
    font-size: 12px;
  }
}

.Settings__toggle.on {
  .Settings__toggleSlider {
    background-color: $opaque-ghosty-blue;
  }

  .Settings__toggleButton {
    left: 11px;
    background-color: $ghosty-blue;
  }

  > p {
    color: $ghosty-blue;
  }
}

.Settings__blueNotificationText {
  margin: 2px 0 14px;
  font-weight: 400;
  font-size: 12px;
  text-decoration: underline;
  color: $ghosty-blue;
  cursor: pointer;

  &.disabled {
    text-decoration: none;
    color: $medium-grey;
    pointer-events: none;
  }
}

.Settings__comingSoon {
  margin: 7px 0;
  width: 100%;
  font-size: 12px;
  text-align: center;
  color: $dark-grey;
}

.Settings__footer {
  background-color: $opaque-background-white;
  border-top: 1px solid $dark-grey;
  height: $footer-height;
}

.Settings__accountBadge {
  width: 100%;
  height: 60px;

  &.gradient {
    background: $background-button-gradient;
  }
}

.Settings__insightsBadge {
  margin: 18px 38.6px 0 13px;
  width: 59.4px;
  height: 29.6px;

  &.white {
    margin-top: 0;
  }
}

.Settings__badgeText {
  font-weight: 700;
  font-size: 13px;
  line-height: 22px;
  text-align: left;
  color: $white;
}

.Settings__footerSettingContainer {
  margin: 17px 0 0 13px;

  &.clickable {
    cursor: pointer;
  }
}

.Settings__footerIcon {
  margin-right: 13px;
  width: 18px;
  height: 18px;
}

.Settings__footerText {
  margin: 0;
  font-weight: 700;
  font-size: 13px;
  color: $dark-grey;
}

.Settings__signOut {
  margin: 43px 0 43px 13px;
  font-style: normal;
  color: $dark-grey;
  cursor: pointer;
}

.Settings__secondaryPage {
  padding-left: 13px;
  width: 100%;
  font-size: 12px;
  text-align: left;
  color: $dark-grey;
}

.Settings__backButton {
  position: relative;
  top: 8px;
  left: -2px;
  margin-bottom: 30px;
  width: 60px;
  cursor: pointer;

  img {
    transform: rotate(90deg);
    margin: 0 4px 0 0;
    width: 12px;
    height: 6.5px;
  }

  p {
    margin: 0;
  }
}

.Settings__primaryPageLink {
  font-size: 12px;
  text-decoration: none;
  color: $ghosty-blue;

  &:hover {
    cursor: pointer;
  }
}

.Settings__secondaryPageTitle {
  width: 260px;
  font-weight: 700;
  font-size: 13px;
}

.Settings__secondaryPageInternalLink {
  color: $ghosty-blue;
  cursor: pointer;
}

.Settings__secondaryPageLink {
  text-decoration: none;
  color: $ghosty-blue;
}

.Settings--smallVerticalMargin {
  margin: 3px 0;
}

.Settings--marginTop {
  margin-top: 30px;
}
