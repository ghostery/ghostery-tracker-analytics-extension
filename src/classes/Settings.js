/**
 * Settings Class
 *
 * _settings acts as a cache for lookup, preventing a call to chrome.storage.local
 *
 * Insights by Ghostery
 */

import cloneDeep from 'lodash.clonedeep';
import isObject from 'lodash.isobject';
import { log } from '../utils/common';

export const DEFAULT_SETTINGS = {
  starredTrackerIds: {},
  pageEvents: {},
  blueBarPosition: 'default',
  showBlueBar: false,
  showToasts: true,
  timelineScale: 'linear',
  infoCenterShown: false,
  tabIconExplainerAcknowledged: false,
  freeTrialStatus: null,
  endOfFreeTrialPopupAcknowledged: false,
  bugs: {},
  metrics: {},
};

class Settings {
  constructor() {
    this._settings = cloneDeep(DEFAULT_SETTINGS);
    this._initComplete = false;
    this._preInitProps = {};
  }

  show = () => {
    const { _settings, ...otherProps } = this;
    log('Settings', _settings, JSON.stringify(otherProps));
    chrome.storage.local.get(null, log);
  }

  // The Default Settings are already loaded via the Constructor.
  // This function overrides the Default Settings with Locally Stored Settings.
  // Those settings are then overwritten by Account Synched Settings.
  init() {
    return new Promise((resolve, reject) => {
      this.initFromLocalStorage()
        .then(this.initFromAccountServer)
        .then(this.completeInit)
        .then(resolve)
        .catch(reject);
    });
  }

  // Overwrite Settings with Locally Stored Settings
  initFromLocalStorage() {
    return new Promise((resolve, reject) => {
      const settings = this._settings;
      chrome.storage.local.get(null, (items) => {
        if (chrome.runtime.lastError) {
          reject();
        }
        for (const prop in items) {
          if (items.hasOwnProperty(prop) && settings.hasOwnProperty(prop)) {
            settings[prop] = items[prop];
          }
        }
        chrome.storage.local.clear();
        chrome.storage.local.set(settings);
        resolve(settings);
      });
    });
  }

  // ToDo: Overwrite Settings with Account Synched Settings
  initFromAccountServer() {
    return new Promise((resolve) => {
      resolve();
    });
  }

  completeInit = () => new Promise((resolve, reject) => {
    this.set(this._preInitProps).then(() => {
      this._initComplete = true;
      this._preInitProps = {};
      resolve();
    }).catch(reject);
  });

  // TODO always return an object, even when just a single property was requested
  get(props) {
    if (typeof props === 'undefined') {
      return this._settings;
    }
    if (typeof props === 'string') {
      return this._getSingleProp(props);
    }
    if (Array.isArray(props)) {
      const propValues = {};
      for (let i = 0; i < props.length; i++) {
        const prop = props[i];
        const value = this._getSingleProp(prop);
        if (value) {
          propValues[prop] = value;
        }
      }
      return propValues;
    }

    return undefined;
  }

  _getSingleProp(property) {
    if (this._settings.hasOwnProperty(property)) {
      return this._settings[property];
    }
    return undefined;
  }

  set(props, value) {
    const propValues = {};
    if (typeof props === 'string' && value !== 'undefined') {
      propValues[props] = value;
    }
    if (isObject(props)) {
      for (const key in props) {
        if (props.hasOwnProperty(key)) {
          propValues[key] = props[key];
        }
      }
    }
    return this._setProps(propValues);
  }

  _setProps(props) {
    return new Promise((resolve, reject) => {
      const setProps = {};
      for (const key in props) {
        if (props.hasOwnProperty(key) && this._settings.hasOwnProperty(key)) {
          this._settings[key] = props[key];
          setProps[key] = props[key];
        }
      }

      if (this._initComplete) {
        chrome.storage.local.set(setProps, () => {
          if (chrome.runtime.lastError) {
            reject();
          }
          resolve(setProps);
        });
      } else {
        this._preInitProps = setProps;
        resolve(setProps);
      }
    });
  }
}

export default new Settings();
