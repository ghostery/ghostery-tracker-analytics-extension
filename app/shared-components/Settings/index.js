import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Settings from './Settings.jsx';
import {
  resetBlueBar,
  login,
  logout,
  openAccountPage,
  sendMetrics,
  openProductTour,
  openGlossary,
  openLicenses,
  toggleBlueBar,
  toggleToasts,
} from '../../store/messages/messageCreators';
import { closeToast } from '../../store/actions/UserInterfaceActions';

const mapStateToProps = state => ({
  userInfo: state.SettingsReducer.userInfo,
  showBlueBar: state.SettingsReducer.localSettings.showBlueBar,
  showToasts: state.SettingsReducer.localSettings.showToasts,
  messageCreators: {
    resetBlueBar,
    login,
    logout,
    openAccountPage,
    sendMetrics,
    openProductTour,
    openGlossary,
    openLicenses,
    toggleBlueBar,
    toggleToasts,
  },
  isPageNotScanned: state.PageInfoReducer.isPageNotScanned,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ closeToast }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
