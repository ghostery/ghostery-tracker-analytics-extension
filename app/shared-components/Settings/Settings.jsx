import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import Tab from 'react-bootstrap/Tab';
import Nav from 'react-bootstrap/Nav';

import Globals from '../../../src/classes/Globals';
import FavoriteTrackersSearch from './FavoriteTrackersSearch';
import './Settings.scss';

const {
  EXTENSION_VERSION,
} = Globals;


class Settings extends Component {
  constructor(props) {
    super(props);
    const { messageCreators } = props;
    this.state = {
      view: 'mainTabs',
      viewLinks: {
        about: [
          { pos: 1, type: 'title', text: 'About' },
          { pos: 2, type: 'text', text: 'Ghostery Insights Beta for Chrome' },
          { pos: 3, type: 'text', text: `Version ${EXTENSION_VERSION}` },
          { pos: 4, type: 'internalLink', text: 'Licenses', handler: messageCreators.openLicenses },
          { pos: 5, type: 'link', text: 'Privacy Policy', link: 'https://www.ghostery.com/about-ghostery/ghostery-plans-and-products-privacy-policy/' },
          { pos: 6, type: 'link', text: 'End-User License Agreement', link: 'https://www.ghostery.com/about-ghostery/ghostery-subscription-products-end-user-license-agreement/' },
          { pos: 7, type: 'link', text: 'Imprint', link: 'https://www.ghostery.com/about-ghostery/imprint/' },
          { pos: 8, type: 'link', text: 'Ghostery Website', link: 'https://www.ghostery.com/' },
        ],
        support: [
          { pos: 1, type: 'title', text: 'Submit a bug', additionalClass: 'Settings--marginTop' },
          { pos: 2, type: 'link', text: 'support-insights@ghostery.com', link: 'mailto:support-insights@ghostery.com', disableTarget: true },
        ],
      },
    };
  }

  handleClickSetView = (view) => {
    this.setState({ view });
  }

  render() {
    const {
      closeDropdown,
      panel,
      messageCreators,
      showBlueBar,
      showToasts,
      isPageNotScanned,
    } = this.props;
    const { view, viewLinks } = this.state;
    const defaultPositionTextClasses = ClassNames(
      'Settings__blueNotificationText',
      { disabled: !showBlueBar },
    );
    const blueBarToggleClasses = ClassNames(
      'Settings__toggle',
      'd-flex',
      'align-items-center',
      { on: showBlueBar },
    );
    const toastsToggleClasses = ClassNames(
      'Settings__toggle',
      'd-flex',
      'align-items-center',
      { on: showToasts },
    );
    const blueBarToggleText = showBlueBar ? 'On' : 'Off';
    const toastToggleText = showToasts ? 'On' : 'Off';
    const displayLinks = viewLinks[view];

    return (
      <React.Fragment>
        {!isPageNotScanned && (
          <div className="Settings">
            {view === 'mainTabs' && (
              <div className="Settings__main d-flex flex-column">
                <img
                  src={[chrome.extension.getURL('dist/images/shared/x-icon.svg')]}
                  className="Settings__closeButton"
                  alt="Star icon"
                  onClick={closeDropdown}
                />
                <div className="Settings__settingsTabsContainer d-flex flex-column justify-content-between">
                  <Tab.Container defaultActiveKey={1}>
                    <div className="Settings__settingsTabs d-flex flex-column">
                      <div className="d-flex">
                        <div className="Settings__blackLine" />
                        <Nav>
                          <Nav.Item key="display">
                            <Nav.Link className="Settings__settingsTab" eventKey={1}>
                              Settings &amp; Help
                            </Nav.Link>
                          </Nav.Item>
                          <Nav.Item key="favorite">
                            <Nav.Link className="Settings__settingsTab" eventKey={2}>
                              Favorite Trackers
                            </Nav.Link>
                          </Nav.Item>
                        </Nav>
                        <div className="Settings__blackLine" />
                      </div>
                      <Tab.Content className="d-flex flex-column">
                        <Tab.Pane className="Settings__tabPane" key="display" eventKey={1}>
                          <div className="Settings__displaySettingsTabContent d-flex flex-column align-items-start">
                            <div className="Settings__textToggleContainer d-flex justify-content-between align-items-center">
                              <p className="Settings__headerText">Blue Summary Bar</p>
                              <div
                                className={blueBarToggleClasses}
                                onClick={() => messageCreators.toggleBlueBar(!showBlueBar)}
                              >
                                <div className="Settings__toggleSlider" />
                                <div className="Settings__toggleButton" />
                                <p>{blueBarToggleText}</p>
                              </div>
                            </div>
                            <p
                              className={defaultPositionTextClasses}
                              onClick={() => messageCreators.resetBlueBar('default')}
                            >
                              Reset to default position
                            </p>
                            <div className="Settings__textToggleContainer d-flex justify-content-between align-items-center">
                              <p className="Settings__headerText">Pop-Up Notifications</p>
                              <div
                                className={toastsToggleClasses}
                                onClick={() => messageCreators.toggleToasts(!showToasts)}
                              >
                                <div className="Settings__toggleSlider" />
                                <div className="Settings__toggleButton" />
                                <p>{toastToggleText}</p>
                              </div>
                            </div>
                            <div
                              className="Settings__primaryPageLink Settings--smallVerticalMargin"
                              onClick={() => messageCreators.openProductTour()}
                            >
                              Open Product Tour
                            </div>
                            <div
                              className="Settings__primaryPageLink Settings--smallVerticalMargin"
                              onClick={() => messageCreators.openGlossary()}
                            >
                              Open Glossary
                            </div>
                            <a className="Settings__primaryPageLink Settings--smallVerticalMargin" href="https://www.surveymonkey.com/r/2J3P5CG" target="_blank" rel="noopener noreferrer">
                              Give us feedback
                            </a>
                          </div>
                        </Tab.Pane>
                        <Tab.Pane className="Settings__tabPane" key="favorite" eventKey={2}>
                          <FavoriteTrackersSearch panel={panel} />
                        </Tab.Pane>
                      </Tab.Content>
                    </div>
                  </Tab.Container>
                  <div className="Settings__footer d-flex flex-column align-items-start">
                    <div
                      className="Settings__footerSettingContainer d-flex justify-content-start clickable"
                      onClick={() => this.handleClickSetView('about')}
                    >
                      <img
                        alt="About"
                        src={[chrome.extension.getURL('/dist/images/shared/info-icon-settings.svg')]}
                        className="Settings__footerIcon"
                      />
                      <p className="Settings__footerText">About</p>
                    </div>
                  </div>
                </div>
              </div>
            )}
            {displayLinks && (
              <div className="Settings__secondaryPage">
                <div
                  className="Settings__backButton d-flex align-items-center"
                  onClick={() => this.handleClickSetView('mainTabs')}
                >
                  <img
                    className="Settings__backArrow"
                    alt="Go back to Settings main menu"
                    src={[chrome.extension.getURL('/dist/images/shared/caret-down-dark.svg')]}
                  />
                  <p>Back</p>
                </div>
                <div className="Settings__secondaryPageContent d-flex flex-column align-items-start">
                  {displayLinks.map((linkObj) => {
                    switch (linkObj.type) {
                      case 'title': {
                        const titleClasses = ClassNames(
                          'Settings__secondaryPageTitle',
                          [linkObj.additionalClass],
                        );
                        return (
                          <p key={linkObj.pos} className={titleClasses}>
                            {linkObj.text}
                          </p>
                        );
                      }
                      case 'text': {
                        return (
                          <p key={linkObj.pos} className="Settings--smallVerticalMargin">
                            {linkObj.text}
                          </p>
                        );
                      }
                      case 'link': {
                        return (
                          <a
                            key={linkObj.pos}
                            className="Settings__secondaryPageLink Settings--smallVerticalMargin"
                            href={linkObj.link}
                            target={!linkObj.disableTarget ? '_blank' : undefined}
                            rel="noopener noreferrer"
                          >
                            {linkObj.text}
                          </a>
                        );
                      }
                      case 'internalLink': {
                        return (
                          <p
                            key={linkObj.pos}
                            className="Settings__secondaryPageInternalLink Settings--smallVerticalMargin"
                            onClick={(e) => {
                              e.stopPropagation();
                              linkObj.handler();
                            }}
                          >
                            {linkObj.text}
                          </p>
                        );
                      }
                      default: {
                        return (<div key={linkObj.pos} />);
                      }
                    }
                  })}
                </div>
              </div>
            )}
          </div>
        )}
      </React.Fragment>
    );
  }
}

Settings.propTypes = {
  panel: PropTypes.bool,
  closeDropdown: PropTypes.func.isRequired,
  userInfo: PropTypes.shape({
    freeTrial: PropTypes.bool.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
  showBlueBar: PropTypes.bool.isRequired,
  showToasts: PropTypes.bool.isRequired,
  messageCreators: PropTypes.shape({
    toggleBlueBar: PropTypes.func.isRequired,
    resetBlueBar: PropTypes.func.isRequired,
    toggleToasts: PropTypes.func.isRequired,
    logout: PropTypes.func.isRequired,
    sendMetrics: PropTypes.func.isRequired,
    openAccountPage: PropTypes.func.isRequired,
    openProductTour: PropTypes.func.isRequired,
    openGlossary: PropTypes.func.isRequired,
    openLicenses: PropTypes.func.isRequired,
  }).isRequired,
  actions: PropTypes.shape({
    closeToast: PropTypes.func.isRequired,
  }).isRequired,
  isPageNotScanned: PropTypes.bool.isRequired,
};

Settings.defaultProps = {
  panel: false,
};

export default Settings;
