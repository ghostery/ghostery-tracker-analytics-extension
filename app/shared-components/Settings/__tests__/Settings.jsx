import 'jsdom-global/register';
import React from 'react';
import { shallow } from 'enzyme';
import path from 'path';
import fakeProps from 'react-fake-props';
import chrome from 'sinon-chrome';

import Settings from '../Settings';

jest.mock('../../../../src/classes/Globals', () => ({
  BROWSER_INFO: { displayName: 'Chrome' },
}));

describe('app/shared-components/Settings/', () => {
  describe('Snapshot tests with shallow', () => {
    global.chrome = chrome;
    test('Settings component should render', () => {
      const componentPath = path.join(__dirname, '../Settings.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<Settings {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
