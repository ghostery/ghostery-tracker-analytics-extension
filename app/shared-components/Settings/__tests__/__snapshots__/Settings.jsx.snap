// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/Settings/ Snapshot tests with shallow Settings component should render 1`] = `<Fragment />`;
